File: services/cms-svc/src/main/java/com/supportme/pay/cmssvc/beneficiary/BeneficiaryNodeRepository.java
package com.supportme.pay.cmssvc.beneficiary;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface BeneficiaryNodeRepository extends JpaRepository<BeneficiaryNode, Long> {
    List<BeneficiaryNode> findByOrganizationIdOrderByPosition(Long organizationId);

    List<BeneficiaryNode> findByOrganizationIdAndActiveTrueOrderByPosition(Long organizationId);
}

File: pay-backend/storefront-domain/src/main/kotlin/com/supportme/pay/storefront/domain/repository/BeneficiaryNodeRepository.kt
package com.supportme.pay.storefront.domain.repository

import com.supportme.pay.storefront.domain.entity.BeneficiaryNode
import org.springframework.data.jpa.repository.JpaRepository

interface BeneficiaryNodeRepository : JpaRepository<BeneficiaryNode, Long> {
    fun findAllByActiveTrueOrderByPositionAscIdAsc(): List<BeneficiaryNode>

    fun findAllByOrderByPositionAscIdAsc(): List<BeneficiaryNode>
}
